src/payment/data/service.js

```diff
@@ -32,6 +33,13 @@ export function configureApiService(newConfig, newApiClient) {
   apiClient.interceptors.response.use((response) => {
     const { status, data } = response;
     if (status >= 200 && status < 300 && data && data.redirect) {
+      // Redirecting this SPA to itself is likely to cause
+      // a redirect loop.
+      if (window.location.href === data.redirect) {
+        logError('The api response is directing the basket page to redirect to itself', {
```

Comment:
Maybe: `'The api response is redirecting to the same payment page url'`?

Reply:
"Houston, we have a loop."